mbed_host_tests/host_tests_runner/host_test_default.py

```diff
@@ -247,6 +251,29 @@ def start_conn_process():
                         else:
                             self.logger.prn_err("orphan event in preamble phase: {{%s;%s}}, timestamp=%f"% (key, str(value), timestamp))
                     else:
+                        # If coverage detected use idle loop
+                        if key == '__coverage_start':
+                            if not coverage_idle:
+                                self.logger.prn_inf("starting coverage idle timeout loop...")
+                                coverage_idle = True
+
+                                # Save position in timeout to resume after coverage
+                                resume_timeout = timeout_duration - (time() - start_time)
```

Comment:
As discussed changing loop variables within the loop can be difficult to understand by some. It would be a good Idea to branch out into a different function that has the other loop driven by idle timeout for processing coverage data. After returning from that function the time taken by that function can be added to ```timeout_duration``` to accommodate time added by coverage data.